addons/website_sale/models/website.py

```diff
@@ -161,8 +161,10 @@ def _prepare_sale_order_values(self, partner, pricelist):
         affiliate_id = request.session.get('affiliate_id')
         salesperson_id = affiliate_id if self.env['res.users'].sudo().browse(affiliate_id).exists() else request.website.salesperson_id.id
         addr = partner.address_get(['delivery', 'invoice'])
-        if len(partner.sale_order_ids):  # first = me
-            addr['delivery'] = partner.sale_order_ids[0].partner_shipping_id.id
+        if not request.website.is_public_user():
+            last_sale_order = self.env['sale.order'].sudo().search([('partner_id', '=', partner.id)], limit=1)
```

Comment:
Why is required sudo here if before your change was not required?

sudo is not bypassing multicompany or others ACLs?
You could get a sale that the user doesn’t have permission to see.

The previous code is using [0] for the first record
But you are naming your variable “last” that means that the behaviour was changed?

Shouldn’t you send the explicit ”order by” in order to avoid a re-definition of _order from another module that changes your current expected result

Reply:
Hello thanks for noticing this issues, I have removed the `sudo` and explicitly defined the order.

Using `[0]` was actually retrieving the last sale order since the order is set to retrieve the last sale order first.